app/controllers/vendors_controller.rb

```diff
@@ -1,7 +1,11 @@
 class VendorsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :authorize_user!, only: [:update, :destory]
```

Comment:
Yes. Admins are going to approve the vendors and set vendor owners; the vendor owners will be the one who can edit vendor specific information (name, category, details).